phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/containers/ContainerVerbsTab.java

```diff
@@ -0,0 +1,94 @@
+package org.phoenicis.javafx.views.mainwindow.containers;
+
+import javafx.application.Platform;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.Tab;
+import javafx.scene.layout.TilePane;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+import org.phoenicis.containers.dto.ContainerDTO;
+import org.phoenicis.engines.VerbsManager;
+import org.phoenicis.javafx.views.common.ErrorMessage;
+import org.phoenicis.javafx.views.common.TextWithStyle;
+import org.phoenicis.repository.dto.ApplicationDTO;
+import org.phoenicis.repository.dto.ScriptDTO;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.phoenicis.configuration.localisation.Localisation.tr;
+
+public class ContainerVerbsTab extends Tab {
+    private static final String CONFIGURATION_PANE_CSS_CLASS = "containerConfigurationPane";
+    private static final String TITLE_CSS_CLASS = "title";
+
+    private final ContainerDTO container;
+    private VerbsManager verbsManager;
+
+    private final List<Node> lockableElements = new ArrayList<>();
+
+    public ContainerVerbsTab(ContainerDTO container, VerbsManager verbsManager,
+            ApplicationDTO engineTools) {
+        super(tr("Verbs"));
+
+        this.container = container;
+        this.verbsManager = verbsManager;
+
+        this.setClosable(false);
+
+        this.populate(engineTools);
+    }
+
+    private void populate(ApplicationDTO verbs) {
+        final VBox verbsPane = new VBox();
+        final Text title = new TextWithStyle(tr("Verbs"), TITLE_CSS_CLASS);
+
+        verbsPane.getStyleClass().add(CONFIGURATION_PANE_CSS_CLASS);
+        verbsPane.getChildren().add(title);
+
+        final TilePane verbsContentPane = new TilePane();
+        verbsContentPane.setPrefColumns(3);
+        verbsContentPane.getStyleClass().add("grid");
+
+        for (ScriptDTO verb : verbs.getScripts()) {
+            Button verbButton = new Button(verb.getScriptName());
+            verbButton.getStyleClass().addAll("toolButton");
+            verbButton.setStyle("-fx-background-image: url('" + verb.getIcon() + "');");
+            verbButton.setOnMouseClicked(event -> {
+                this.lockAll();
+                // TODO: find a better way to get the engine ID
+                this.verbsManager.installVerb(container.getEngine().toLowerCase(), container.getName(), verb.getId(),
+                        this::unlockAll,
+                        e -> Platform.runLater(() -> new ErrorMessage("Error", e).show()));
+            });
+            this.lockableElements.add(verbButton);
+            verbsContentPane.getChildren().add(verbButton);
+        }
+
+        verbsPane.getChildren().add(verbsContentPane);
+
+        final ScrollPane scrollPane = new ScrollPane();
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER); // the TilePane adjusts the number of columns
```

Comment:
This line should go to the css files

Reply:
That's just copy paste from the engine tools.

Reply:
Do you think that some of the code would be factored?

Reply:
Any suggestions/what would you expect?

Reply:
@qparis I'm not sure what you mean?